### PR TITLE
ENH: remove variable from torch doctests, make functionals work

### PR DESCRIPTION
**Pull request by kohr-h**
- Adapt doctests to modern PyTorch, i.e., remove `Variable` and update printing. Now they pass again.
- Make functionals work as `Module`, handy for using custom loss functions.

### CODE REVIEW

File: odl/contrib/torch/operator.py [hunk 1]
```diff
@@ -329,51 +310,39 @@ def forward(self, x):
...
-        [torch.FloatTensor of size 3x2x2]
+                [[ 4.,  8.],
+                 [ 5., 10.]]])
         """
         in_shape = x.data.shape
-        op_in_shape = self.op_func.operator.domain.shape
-        op_out_shape = self.op_func.operator.range.shape
+        op_in_shape = self.operator.domain.shape
+        if self.operator.is_functional:
+            op_out_shape = ()
```
Comment (adler-j) [inline odl/contrib/torch/operator.py]: We could give `Field` a shape `()` to avoid these special cases, if we wanted to.

Comment (kohr-h) [reply odl/contrib/torch/operator.py]: Yes, indeed. It's also related to #1435
